core/drivers/crypto/se050/crypto.mk

```diff
@@ -0,0 +1,67 @@
+ifeq ($(CFG_NXP_SE05X), y)
+# Enable the crypto driver
+CFG_CRYPTO_DRIVER = y
```

Comment:
no, you cant force on the driver specific code without changes in the platform independent makefile (and you dont want to do that)
also  CFG_CRYPTO_DRIVER ?= y makes no sense (what would be the point of it? this is the crypto driver we are trying to build....). it must be y.

Reply:
So it must be `y` but you don't want to force it `y`.    **:\\**
`CFG_CRYPTO_DRIVER = y` here makes no guarantee it will be set to `y` hence the proposed use of `$(call force, ...)`.

Reply:
@etienne-lms I can see I am failing to explain the issue with that apparently logical approach you are requesting (I am perhaps being too generic, sorry about that). Let me try with an example:

If we do what you propose, the user has to add CFG_CRYPTO_DRIVER = y to the build line. So far so good. 

Now say the user wants to run on an IMX platform. The build framework will execute the make script in plat-imx/crypto_conf.mk .

IMX platforms already implement cryptographic devices (CAAM) on most SoCs. And as far as the platform is concerned, since the user didnt select this CAAM device (or any other cryptographic driver for that matter), it makes sense for the platform to force CFG_CRYPTO_DRIVER to 'n'; this is because the platform is supposed to be generic IMO. 

And of course, the build will fail. You see the problem?

So of course, we could add SE50 specific configs to _all_ platforms and therefore over-complicate the already messy build scripts.

Or we can just do what I propose and let the driver script that executes _after_ the platform overwrite the platform choices.

Reply:
The issue is that `CFG_CRYPTO_DRIVER = y` here produces no guaranty `CFG_CRYPTO_DRIVER` is set to `y`.
I think the issue is in **plat-imx** and **plat-ls** that consider `CFG_CRYPTO_DRIVER=y` is used only for CAAM.
I think these platforms should no force `CFG_CRYPTO_DRIVER` to `n` when CAAM is disabled.

Reply:
I fully disagree with you but sure, let me know what I should do to get this merged.

Reply:
> you are saying that: platforms **must not force** configuration options that could be required by drivers that are platform independent.
--> and now every platform needs to know about all op-tee drivers that might be using those options

I think imx and ls paltforms should not presume `CFG_CRYPTO_DRIVER` is meant for CAAM only if they want to optionally embed SE050. A platform that forces `CFG_CRYPTO_DRIVER` to `n` obviosuly can't enable SE050.

Reply:
right but why should they even know that the particular board will have a crypto device on some i2c bus? as far as the platform is concerned, they can force that the soc will not have a platform crypto driver (or any other platform feature for that matter). And if the board then wants to include one, let the board override whatever the platform assume by using the driver's crypto make. I think this scales better IMO. 

But as I said, if you dont want to go that route, just let me know and I'll fix and test on IMX and deliver the new patches (just looking forward to get this upstream and add the missing ECC functionality)

Reply:
We tend to be picky with `CFG_` configs so that it is not to messy and possibly evolve to some more standard config framework hence default/forced exclusively from config scripts.
@jforissier, what do you think about using `CFG_xxx = y/n` in a configuration script? Does it make sense to you?

Reply:
@etienne-lms you're right, only `?=` or `$(call force...)` to set `CFG_` is the rule. The reason is (1) allow users to provide parameters either via environment variables or the make command line, with a consistent behavior (`CFG_FOO=y make` vs. `make CFG_FOO=y` are equivalent), and (2) error out if a user-supplied parameter (env var or command line) conflicts with a mandatory value (hence the `force` macro). `CFG_XXX = y/n` is problematic on both counts.

In this particular case I believe the platform should not set `CFG_CRYPTO_DRIVER` (that is, leave it undefined, **not** set it to `n`), and then `crypto.mk` should `$(call force,CFG_CRYPTO_DRIVER,y)` if `CFG_NXP_SE05X` is `y`.

Reply:
ok. pushing an updated PR.